src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolver.cs

```diff
@@ -0,0 +1,276 @@
+        }
+
+        public List<string> GetWorkerConfigs(List<string> probingPaths, string fallbackPath)
+        {
+            // Dictionary of { FUNCTIONS_WORKER_RUNTIME environment variable value : path of workerConfig }
+            Dictionary<string, string> outputDict = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            var workerRuntime = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionWorkerRuntime);
+            string releaseChannel = Utility.GetPlatformReleaseChannel(_environment);
+
+            if (!probingPaths.IsNullOrEmpty())
+            {
+                // probing path directory structure is: <probingPath>/<workerRuntime>/<version>/<worker.config.json>
+                foreach (var probingPath in probingPaths)
+                {
+                    if (!string.IsNullOrEmpty(probingPath) && Directory.Exists(probingPath))
+                    {
+                        foreach (var languageWorkerPath in Directory.EnumerateDirectories(probingPath))
+                        {
+                            string languageWorkerDir = Path.GetFileName(languageWorkerPath);
+
+                            // If probing paths are malformed and have duplicate directories of the same language worker (eg. due to different casing)
+                            if (outputDict.ContainsKey(languageWorkerDir))
+                            {
+                                continue;
+                            }
+
+                            // Only skip worker directories that don't match the current runtime or are not enabled via hosting config
+                            if ((_workersAvailableForResolutionViaHostingConfig is not null &&
+                                !_workersAvailableForResolutionViaHostingConfig.Contains(languageWorkerDir)) ||
+                                (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                                !_environment.IsPlaceholderModeEnabled() &&
+                                workerRuntime is not null &&
+                                !workerRuntime.Equals(languageWorkerDir, StringComparison.OrdinalIgnoreCase)))
+                            {
+                                continue;
+                            }
+
+                            IEnumerable<string> workerVersions = Directory.EnumerateDirectories(languageWorkerPath);
+                            var versions = GetWorkerVersionsDescending(workerVersions);
+
+                            outputDict = GetWorkerConfigsFromProbingPaths(versions, languageWorkerPath, languageWorkerDir, releaseChannel, outputDict);
+                        }
+                    }
+                }
+            }
+
+            if (!_environment.IsMultiLanguageRuntimeEnvironment() &&
+                workerRuntime is not null &&
+                outputDict.ContainsKey(workerRuntime))
```

Comment:
one confusing scenario is -- what if someone has the releasechannel set to Standard yet we only have a single release in this path? would we go back to the fallback to find out what else is there?

Reply:
If we have a single release in this path and it satisfies the compatibility check then we will return that for the Standard release channel setting too.